### PR TITLE
Bumps Go to 1.17 and CAPI to v1.1.0

### PR DESCRIPTION
**Pull request by sayantani11**
**What type of PR is this?**

**What this PR does / why we need it**:

- Bumps CAPI to v1.1.0
- Bumps Go to 1.17

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Update to v1.1.0
```

### CODE REVIEW

**Comment by cpanato** [comment]
@sayantani11 please rebase your PR with latest main branch, and looks like some updates need some changes

**Comment by fabriziopandini** [comment]
Thanks for picking up the new release so fast!

**Comment by cpanato** [comment]
@sayantani11, we need to bump the project to use go 1.17. Do you want to do that? i can do and then you can see as example, but if you want to do go for it :) can be in this same PR, but we will need to update the title and description later

let me know what are your thoughts

**Comment by sayantani11** [comment]
> @sayantani11, we need to bump the project to use go 1.17. Do you want to do that? i can do and then you can see as example, but if you want to do go for it :) can be in this same PR, but we will need to update the title and description later
> 
> let me know what are your thoughts

Let me give it a try once🙂

**Comment by cpanato** [comment]
@sayantani11 @jayesh-srivastava there two similar PRs, this and https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/504 can you both sync and work together?
let me know if need any help

**Comment by jayesh-srivastava** [comment]
@cpanato Sure!

**Comment by sayantani11** [comment]
@cpanato Can you please help with the CI jobs? I am not able to understand what's wrong.

File: Dockerfile [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 # Build the manager binary
-FROM golang:1.16.11 as builder
+FROM golang:1.17 as builder
```
Comment (cpanato) [inline Dockerfile:16]: please use `1.17.6`

File: Makefile [hunk 2]
```diff
@@ -56,35 +56,35 @@ KUBETEST_CONF_PATH ?= $(abspath $(E2E_DATA_DIR)/kubetest/conformance.yaml)
...
 CONTROLLER_GEN_BIN := controller-gen
 CONTROLLER_GEN := $(TOOLS_BIN_DIR)/$(CONTROLLER_GEN_BIN)-$(CONTROLLER_GEN_VER)
 
-CONVERSION_GEN_VER := v0.22.2
+CONVERSION_GEN_VER := v0.23.1
 CONVERSION_GEN_BIN := conversion-gen
 CONVERSION_GEN := $(TOOLS_BIN_DIR)/$(CONVERSION_GEN_BIN)-$(CONVERSION_GEN_VER)
 
-ENVSUBST_VER := v1.2.0
+ENVSUBST_VER := v2.0.0-20210730161058-179042472c46
```
Comment (cpanato) [inline Makefile:67]: keep the current version, lets try to keep the pr small as possible, only if this is really required to be updated

File: Makefile [hunk 3]
```diff
@@ -56,35 +56,35 @@ KUBETEST_CONF_PATH ?= $(abspath $(E2E_DATA_DIR)/kubetest/conformance.yaml)
 # Binaries.
 CLUSTERCTL := $(BIN_DIR)/clusterctl
 
-CONTROLLER_GEN_VER := v0.7.0
+CONTROLLER_GEN_VER := v0.8.0
 CONTROLLER_GEN_BIN := controller-gen
 CONTROLLER_GEN := $(TOOLS_BIN_DIR)/$(CONTROLLER_GEN_BIN)-$(CONTROLLER_GEN_VER)
 
-CONVERSION_GEN_VER := v0.22.2
+CONVERSION_GEN_VER := v0.23.1
```
Comment (cpanato) [inline Makefile:63]: keep the current version, lets try to keep the pr small as possible, only if this is really required to be updated

File: Makefile [hunk 4]
```diff
@@ -56,35 +56,35 @@ KUBETEST_CONF_PATH ?= $(abspath $(E2E_DATA_DIR)/kubetest/conformance.yaml)
 # Binaries.
 CLUSTERCTL := $(BIN_DIR)/clusterctl
 
-CONTROLLER_GEN_VER := v0.7.0
+CONTROLLER_GEN_VER := v0.8.0
```
Comment (cpanato) [inline Makefile:59]: keep the current version, lets try to keep the pr small as possible, only if this is really required to be updated

File: Makefile [hunk 5]
```diff
@@ -56,35 +56,35 @@ KUBETEST_CONF_PATH ?= $(abspath $(E2E_DATA_DIR)/kubetest/conformance.yaml)
...
+ENVSUBST_VER := v2.0.0-20210730161058-179042472c46
 ENVSUBST_BIN := envsubst
 ENVSUBST := $(TOOLS_BIN_DIR)/$(ENVSUBST_BIN)
 
 GOLANGCI_LINT_VER := v1.43.0
 GOLANGCI_LINT_BIN := golangci-lint
 GOLANGCI_LINT := $(TOOLS_BIN_DIR)/$(GOLANGCI_LINT_BIN)-$(GOLANGCI_LINT_VER)
 
-KUSTOMIZE_VER := v3.8.6
+KUSTOMIZE_VER := v4.4.0
```
Comment (cpanato) [inline Makefile:75]: keep the current version, lets try to keep the pr small as possible, only if this is really required to be updated

File: Makefile [hunk 6]
```diff
@@ -56,35 +56,35 @@ KUBETEST_CONF_PATH ?= $(abspath $(E2E_DATA_DIR)/kubetest/conformance.yaml)
...
 KUSTOMIZE_BIN := kustomize
 KUSTOMIZE := $(TOOLS_BIN_DIR)/$(KUSTOMIZE_BIN)-$(KUSTOMIZE_VER)
 
-RELEASE_NOTES_VER := v0.11.0
+RELEASE_NOTES_VER := v0.12.0
```
Comment (cpanato) [inline Makefile:79]: keep the current version, lets try to keep the pr small as possible, only if this is really required to be updated

File: Makefile [hunk 7]
```diff
@@ -56,35 +56,35 @@ KUBETEST_CONF_PATH ?= $(abspath $(E2E_DATA_DIR)/kubetest/conformance.yaml)
...
 RELEASE_NOTES_BIN := release-notes
 RELEASE_NOTES := $(TOOLS_BIN_DIR)/$(RELEASE_NOTES_BIN)-$(RELEASE_NOTES_VER)
 
-GINKGO_VER := v1.16.4
+GINKGO_VER := v1.16.5
```
Comment (cpanato) [inline Makefile:83]: keep the current version, lets try to keep the pr small as possible, only if this is really required to be updated

File: Makefile [hunk 8]
```diff
@@ -56,35 +56,35 @@ KUBETEST_CONF_PATH ?= $(abspath $(E2E_DATA_DIR)/kubetest/conformance.yaml)
...
 GINKGO_BIN := ginkgo
 GINKGO := $(TOOLS_BIN_DIR)/$(GINKGO_BIN)-$(GINKGO_VER)
 
-KUBECTL_VER := v1.22.3
+KUBECTL_VER := v1.23.3
```
Comment (cpanato) [inline Makefile:87]: keep the current version, lets try to keep the pr small as possible, only if this is really required to be updated

File: Makefile [hunk 9]
```diff
@@ -120,7 +120,7 @@ endif
 # Build time versioning details.
 LDFLAGS := $(shell hack/version.sh)
 
-GOLANG_VERSION := 1.16.10
+GOLANG_VERSION := 1.17
```
Comment (cpanato) [inline Makefile:123]: please use 1.17.6

File: hack/ensure-kustomize.sh [hunk 10]
```diff
@@ -19,7 +19,7 @@ set -o nounset
 set -o pipefail
 
 GOPATH_BIN="$(go env GOPATH)/bin/"
-MINIMUM_KUSTOMIZE_VERSION=3.1.0
```
Comment (cpanato) [inline hack/ensure-kustomize.sh:22]: see my comment above

**Review by cpanato — CHANGES_REQUESTED** [review]
see some notes, but looks like some updates need to be checked in the go module

**Comment by cpanato** [comment]
@sayantani11 this issue with ci is the jobs have go 1.16 and not go 1.17 as soon we merge this https://github.com/kubernetes/test-infra/pull/25141 we can retrigger the jobs

File: Makefile (hunk 2 above)
Comment (sayantani11) [reply Makefile]: @cpanato Yes it's better to just go with Go & CAPI, I will do the others in another PR

**Comment by sayantani11** [comment]
@cpanato We need to upgrade to Ginkgo 16.5? I guess that's what I understood from the failed test🤔

**Comment by sayantani11** [comment]
```sh
To silence deprecations that can be silenced set the following environment variable:
  ACK_GINKGO_DEPRECATIONS=1.16.5
Ginkgo ran 1 suite in 2m7.627675549s
Test Suite Failed
```
This is what the failed test looks like!

**Comment by cpanato** [comment]
> ```shell
> To silence deprecations that can be silenced set the following environment variable:
>   ACK_GINKGO_DEPRECATIONS=1.16.5
> Ginkgo ran 1 suite in 2m7.627675549s
> Test Suite Failed
> ```
> 
> This is what the failed test looks like!

this is not the issue, this is ginkgo just showing that those things will be deprecated soon

the issue is that the clusterctl did not find the latest release, I'm checking your changes

**Review by cpanato — COMMENTED** [review]
add the new contract here as well https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/main/test/e2e/data/shared/v1beta1/metadata.yaml 
I think this will fix the issue

@sayantani11

**Comment by cpanato** [comment]
@sayantani11 i will check what is missing and push if something is missing, this PR is now blocking the other PRs to succeed.

hope that is ok for you

**Comment by sayantani11** [comment]
@cpanato Yes perfectly fine for me

**Review by cpanato — APPROVED** [review]
thanks so much for this PR
/approve
/hold for the final tests
/assign @dims because I did a commit :)

**Comment by dims** [comment]
@cpanato @sayantani11 please feel free to remove hold

**Pull request merged**